mux/src/tmux_pty.rs

```diff
@@ -0,0 +1,118 @@
+use crate::{
+    tmux::{RefTmuxRemotePane, TmuxCmdQueue, TmuxDomainState},
+    tmux_commands::SendKeys,
+};
+use flume;
+use portable_pty::{Child, ExitStatus, MasterPty};
+use std::{
+    io::{Read, Write},
+    sync::{Arc, Condvar, Mutex},
+};
+
+pub(crate) struct TmuxReader {
+    rx: flume::Receiver<String>,
+}
+
+impl Read for TmuxReader {
+    fn read(&mut self, mut buf: &mut [u8]) -> std::io::Result<usize> {
+        match self.rx.recv() {
+            Ok(str) => {
+                return buf.write(str.as_bytes());
+            }
+            Err(_) => {
+                return Ok(0);
+            }
+        }
+    }
+}
+
+// A local tmux pane(tab) based on a tmux pty
+#[derive(Debug, Clone)]
+pub(crate) struct TmuxPty {
+    pub domain_id: usize,
+    pub master_pane: RefTmuxRemotePane,
+    pub rx: flume::Receiver<String>,
+    pub cmd_queue: Arc<Mutex<TmuxCmdQueue>>,
+    pub active_lock: Arc<(Mutex<bool>, Condvar)>,
+}
+
+impl Write for TmuxPty {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        let pane_id = {
+            let pane_lock = self.master_pane.lock().unwrap();
+            pane_lock.pane_id
+        };
+        log::info!("pane:{}, content:{:?}", &pane_id, buf);
+        let mut cmd_queue = self.cmd_queue.lock().unwrap();
+        cmd_queue.push_back(Box::new(SendKeys {
+            pane: pane_id,
+            keys: buf.to_vec(),
+        }));
+        TmuxDomainState::kick_off(self.domain_id);
+        Ok(0)
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        Ok(())
+    }
+}
+
+impl Child for TmuxPty {
+    fn try_wait(&mut self) -> std::io::Result<Option<portable_pty::ExitStatus>> {
+        todo!()
+    }
+
+    fn kill(&mut self) -> std::io::Result<()> {
+        todo!()
+    }
+
+    fn wait(&mut self) -> std::io::Result<portable_pty::ExitStatus> {
+        let (lock, var) = &*self.active_lock;
+        let mut released = lock.lock().unwrap();
+        while !*released {
+            released = var.wait(released).unwrap();
+        }
+        return Ok(ExitStatus::with_exit_code(0));
+    }
+
+    fn process_id(&self) -> Option<u32> {
```

Comment:
There's a windows specific method to impl here too, that can return `None`:

https://github.com/wez/wezterm/blob/3f6ff534d3ad3df8bcf93dd02e1dd6dc3da1d7c2/pty/src/lib.rs#L240-L243